### PR TITLE
fix(dropdown-autocomplete): resolve arrow-key regression caused by addition of react-virtualized

### PR DESCRIPTION
**Pull request by Chrissy**
down-key

tldr;

* autocomplete counts the number of child items and uses this to determine the limit of the index when arrowing around
* react-virtualized will change the order and number of children fed into autocomplete
* this is generally ok because we keep track of things by an index that is assigned from _above_ virtualized, so virtualized can do it's job in ignorant bliss
* except for the arrow key limit, which is not fed in from above but is instead determined at the autocomplete level, so therefore the limit will more or less cap out at 10. 

here is a diagram:

<pre>
dropdownAutocomplete (100 items)
|-react-virtualized(10/100 items)
  |-autocomplete(10 items)
</pre>

to fix this we feed in the results count from the top (omitting labels):

down-key-fixed

### CODE REVIEW

File: src/sentry/static/sentry/app/components/autoComplete.jsx [hunk 1]
```diff
@@ -221,11 +221,11 @@ class AutoComplete extends React.Component {
   };
 
   moveHighlightedIndex = (step, e) => {
-    const listSize = this.items.size - 1;
     let newIndex = this.state.highlightedIndex + step;
+    const listSize = this.itemCount || this.items.size;
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/components/autoComplete.jsx]: When would `this.itemCount` be different to `this.items.size`?

File: src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx [hunk 2]
```diff
@@ -380,6 +380,8 @@ class DropdownAutoCompleteMenu extends React.Component {
           // emptyHidesInput is set to true.
           const showInput = !hideInput && (hasItems || !emptyHidesInput);
 
+          const itemCount = autoCompleteResults.filter(i => !i.groupLabel).length;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx]: If you had 10 results that included 2 `groupLabel`s, wouldn't this just mean that the last 2 items wouldn't be selectable?

Comment (billyvg) [reply src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx]: Disregard, it's because `groupLabels` do not get added to `AutoComplete.items` because of `this.renderRow()` where only autocomplete items get called with `getItemProp` render prop.

Comment (billyvg) [reply src/sentry/static/sentry/app/components/dropdownAutoCompleteMenu.jsx]: So I would add an inline comment here about why we're doing this (virtualization).

Additionally, do we only need to calc this when having virtualized rows?

File: src/sentry/static/sentry/app/components/autoComplete.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/autoComplete.jsx]: Let's add a comment here addressing this.

The answer is that `this.items` only contain the rows that are rendered so when they are virtualized you only get a subset of the items. `itemCount` are all rows (except for groupLabels)

**Review by billyvg — APPROVED** [review]
Total personal preference: I think `totalItems` or `totalSelectableItems` is more clear than `itemCount`

It would be great if we had some follow up tests surrounding the virtualized rows.

**Comment by Chrissy** [comment]
ok this is updated with comments and rtg. gonna merge first thing tomorrow.
